pkg/datamover/repository_server_test.go

```diff
@@ -0,0 +1,216 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datamover
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	. "gopkg.in/check.v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	kopiacmd "github.com/kanisterio/kanister/pkg/kopia/command"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/testutil"
+)
+
+type RepositoryServerSuite struct {
+	ctx                context.Context
+	address            string
+	kopiaCacheDir      string
+	kopiaLogDir        string
+	kopiaConfigDir     string
+	tlsDir             string
+	serverHost         string
+	serverUsername     string
+	serverPassword     string
+	repositoryUser     string
+	testUsername       string
+	testUserPassword   string
+	repositoryPassword string
+	repoPathPrefix     string
+	fingerprint        string
+}
+
+var _ = Suite(&RepositoryServerSuite{})
+
+func (rss *RepositoryServerSuite) SetUpSuite(c *C) {
+	// Check if kopia binary exists in PATH
+	if !CommandExists("kopia") {
+		c.Skip("Skipping repository server datamover unit test. Couldn't find kopia binary in the path.")
+	}
+
+	// Setting Up Repository Server Address
+	rss.address = fmt.Sprintf("%s:%s", "https://0.0.0.0", strconv.Itoa(rand.IntnRange(50000, 60000)))
+
+	// Setting Up Repository Server User Access
+	rss.serverUsername = "user@localhost"
+	rss.serverPassword = "testPassword"
+	rss.serverHost = "localhost"
+	rss.testUsername = "testuser"
+	rss.testUserPassword = rand.String(8)
+
+	// Setting Up Repository Access
+	rss.repositoryUser = "repositoryUser"
+	rss.repositoryPassword = rand.String(8)
+	rss.repoPathPrefix = path.Join("kopia-int", time.Now().UTC().Format(time.RFC3339), rand.String(5))
+
+	rss.ctx = context.Background()
+
+	// Setting Up Kopia Cache, Log and Config Dir
+	rss.kopiaCacheDir = kopiacmd.DefaultCacheDirectory
+	rss.kopiaLogDir = kopiacmd.DefaultLogDirectory
+	rss.kopiaConfigDir = kopiacmd.DefaultConfigDirectory
+
+	// Setting Up TLS Dir
+	temp := c.MkDir()
+	rss.tlsDir = filepath.Join(temp, "tls-"+rand.String(5))
+}
+
+func (rss *RepositoryServerSuite) setupKopiaRepositoryServer(c *C) {
+	// Setting Up Kopia Repository
+	contentCacheMB, metadataCacheMB := kopiacmd.GetGeneralCacheSizeSettings()
+	c.Log("Creating Kopia Repository...")
+	repoCreateCommandArgs := kopiacmd.RepositoryCommandArgs{
+		CommandArgs: &kopiacmd.CommandArgs{
+			RepoPassword:   rss.repositoryPassword,
+			ConfigFilePath: rss.kopiaConfigDir,
+			LogDirectory:   rss.kopiaLogDir,
+		},
+		CacheDirectory:  rss.kopiaCacheDir,
+		Hostname:        rss.serverHost,
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+		RepoPathPrefix:  rss.repoPathPrefix,
+		Username:        rss.repositoryUser,
+		Location:        testutil.GetDefaultS3CompliantStorageLocation(),
+	}
+	repoCreateCmd, err := kopiacmd.RepositoryCreateCommand(repoCreateCommandArgs)
```

Comment:
Would it keep on creating a repository in object store on every run? If yes, should we delete repository in tear down suite? Since it would clutter the bucket with these test repositories

Reply:
I think we don't have Repository Delete Command Wrapper. Might have to add the Wrapper for that

Reply:
There is no command to delete kopia repository, only to disconnect with it.
I've implemented the functionality where we're trying to connect with the Repository first before creating it